### ISSUE TITLE
Messages show up under wrong contact!

### ISSUE THREAD

**Issue by GavinDBrown** [signalapp/Signal-Android#8053]
- [ ] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I see messages from one of my contacts show up correctly on my Android phone but incorrectly as from a different user on the chrome app.

### Steps to reproduce
Note, I have not actually attempted to reproduce this yet, but these are the steps I went through to get to this state.
- Have a Chrome app and Android app linked
- Have a message from Contact 1 be the most recently selected conversation in Chrome app
- Uninstall and reinstall Android app
- Re-link Chrome app
- Receive a message from anyone besides Contact 1.

**Actual result:** Message shows up as from wrong user.
**Expected result:** Gotta get who you're talking to right!

### Screenshots

![signal-bug-phone](https://user-images.githubusercontent.com/5473682/43416166-64732b4c-93ec-11e8-88f2-984ec0caaa0e.png)

![signal-bug-desktop](https://user-images.githubusercontent.com/5473682/43415658-08de0b2c-93eb-11e8-985e-171cb5b2ac52.png)

### Device info
**Device:** Pixel 2
**Android version:** 8.1.0
**Signal version:** 4.23.4

Linux Chrome, latest version.

**Comment by greyson-signal** [comment]
Hi there, while we're looking into this, just wanted to clarify something. When you say "Chrome app", are you referring to the version you installed from the Chrome Web store? If so, that version has been deprecated, and you should instead be using the standalone version that can be found [here](https://signal.org/download/).

**Comment by greyson-signal** [comment]
Also, it looks like there's a chance that "Melody" and "Aunt Gayle" have the same phone number? At least, the last 4 digits appear to be the same? On desktop, can you try going to File > Preferences and click "Import now"? It's possible your contacts are simply out-of-sync.

**Comment by GavinDBrown** [comment]
@greyson-signal 
Yes, I'm using the chrome app, not the standalone desktop version.

The phone number ending in 0383 belongs to Melody, but it incorrectly appears with Aunt Gayle's name. The redacted part in the screenshots is the same.

On the Chrome app I imported contacts and restarted the app, neither resolved the bug.

Happy to help with other debug information if needed :)

**Comment by GavinDBrown** [comment]
No, unfortunately. This is on my work desktop which does not allow external repositories but does allow chrome apps.

**Comment by greyson-signal** [comment]
I don't know how much help we can be then, unfortunately. We don't publish to the Chrome app store anymore because [Google will be removing Mac/Linux/Windows support](https://blog.chromium.org/2016/08/from-chrome-apps-to-web.html) for Chrome desktop apps in the near future. Perhaps you could build the [desktop app](https://github.com/signalapp/Signal-Desktop) from source?

**Comment by GavinDBrown** [comment]
Fair enough. 

I'm still very concerned by this bug though. If it's ever possible to have a conversation listed under the wrong contact that could have extreme consequences for some users. 

I think it's worth confirming that this can exclusively happen with the Chrome app and affirm that the bug is not simply repackaged in the Electron desktop apps. Even then, if this won't get fixed it may be worth pulling the Chrome app now.

**Comment by GavinDBrown** [comment]
That's good to know.

Disabled, re-enabled and restarted Chrome app. Still in the bad state.

**Comment by GavinDBrown** [comment]
Note, I uninstalled then re-installed and re-linked the chrome app and the bad state still persists. I think that's indicative of the corruption also being server-side.

**Comment by haffenloher** [comment]
@GavinDBrown The contact sync messages between your phone and your desktop are end-to-end encrypted. It's not possible for the server to alter their content.

**Comment by GavinDBrown** [comment]
Yet they are mismatched. The contacts on my phone are correct and the contacts on a new install on a desktop are incorrect.

I should note, I used the Chrome app for years without problem before this.

As I see it, either:

1. uninstalling didn't remove the corruption, 
2. the corruption happens when syncing messages, or 
3. the corruption is on my phone and was transferred to the new Chrome app install.

I find 3 unlikely since the contacts correctly appear on my phone.

**Comment by janvlug** [comment]
Could it be that the contacts are somehow linked as being the same contact in Android? If so, this bug might be related to: https://github.com/signalapp/Signal-Desktop/issues/3093.

**Comment by nathan-at-least** [comment]
I have experienced a bug with messages I send to conversation A appearing in the context of conversation B and not on A on my Android Signal Messenger app. Participants in conversation A then reply to my message, so it's apparent they've seen it. I do not believe B participants see the message. Yet Signal continues to display the message in the wrong conversation.

I do not use the Chrome extension.

Should I file a different bug or add to this one? This is the second time I've encountered this bug, and this time I took screenshots of multiple screens to catch the symptoms. Would those be helpful? I would need to redact them.
